app/javascript/src/collected_inks/index.jsx

```diff
@@ -17,495 +17,69 @@ document.addEventListener("DOMContentLoaded", () => {
-          }, [info.rows]);
-          return (
-            <span>
-              <Counter data={counters} field="bottle" />
-              <Counter data={counters} field="sample" />
-              <Counter data={counters} field="cartridge" />
-              <Counter data={counters} field="swab" />
-            </span>
-          );
-        }
-      },
-      {
-        Header: "Color",
-        accessor: "color",
-        Cell: () => "",
-        sortType: sortByColor
-      },
-      {
-        Header: "Swabbed",
-        accessor: "swabbed",
-        Cell: ({ cell: { value } }) => {
-          if (value) {
-            return <i className="fa fa-check" />;
-          } else {
-            return <i className="fa fa-times" />;
-          }
-        },
-        sortType: booleanSort
-      },
-      {
-        Header: "Used",
-        accessor: "used",
-        Cell: ({ cell: { value } }) => {
-          if (value) {
-            return <i className="fa fa-check" />;
-          } else {
-            return <i className="fa fa-times" />;
-          }
-        },
-        sortType: booleanSort
-      },
-      {
-        Header: "Usage",
-        accessor: "usage",
-        sortDescFirst: true
-      },
-      {
-        Header: "Daily Usage",
-        accessor: "daily_usage",
-        sortDescFirst: true
-      },
-      {
-        Header: "Comment",
-        accessor: "comment"
-      },
-      {
-        Header: "Private Comment",
-        accessor: "private_comment"
-      },
-      {
-        Header: "Tags",
-        accessor: "tags",
-        Cell: ({ cell: { value } }) => {
-          if (!value.length) return null;
-          return (
-            <ul className="tags">
-              {value.map((tag) => (
-                <li key={tag.id} className="tag">
-                  {tag.name}
-                </li>
-              ))}
-            </ul>
-          );
-        }
-      }
-    ],
-    []
-  );
-  const hiddenColumns = useMemo(() => {
-    let hidden_columns = [
-      "private_comment",
-      "comment",
-      "maker",
-      "line_name",
-      "kind",
-      "daily_usage"
-    ].filter((n) => !data.some((e) => e[n]));
-    if (data.every((e) => e.tags.length == 0)) hidden_columns.push("tags");
-    return hidden_columns;
-  }, [data]);
-  const {
-    getTableProps,
-    getTableBodyProps,
-    headerGroups,
-    footerGroups,
-    rows,
-    prepareRow,
-    state,
-    preGlobalFilteredRows,
-    setGlobalFilter
-  } = useTable(
-    {
-      columns,
-      data,
-      initialState: {
-        hiddenColumns
-      },
-      filterTypes: {
-        fuzzyText: fuzzyTextFilterFn
-      },
-      globalFilter: "fuzzyText"
+  const storageKey = "fpc-collected-inks-layout";
+  const [layout, setLayout] = useState(storage.getItem(storageKey));
+  const onLayoutChange = useCallback(
+    (e) => {
+      const nextLayout = e.target.value;
+      setLayout(nextLayout);
+      storage.setItem(storageKey, nextLayout);
```

Comment:
I considered using the session cookie for this instead, since `localStorage` is "for ever" (until cleared manually).

That means a user who toggles this (even if only to check what the button does) is now "stuck" in that mode, where a user who never clicked it has a responsive switch based on screen size.

No right answer, but there are other options here.